Lib/typing.py

```diff
@@ -164,14 +164,16 @@ def _type_check(arg, msg, is_argument=True, module=None):
     We append the repr() of the actual value (truncated to 100 chars).
     """
     invalid_generic_forms = (Generic, Protocol)
-    if is_argument:
+    if is_argument and not is_class:
         invalid_generic_forms = invalid_generic_forms + (ClassVar, Final)
+    elif not is_argument and not is_class:  # module / local level
+        invalid_generic_forms = invalid_generic_forms + (ClassVar, )
```

Comment:
This is an overly complicated condition block for what this actually does. How about this instead:

```suggestion
    if not is_class:
        invalid_generic_forms += (ClassVar,)
        if is_argument:
            invalid_generic_forms += (Final,)
```